src/configuration.js

```diff
@@ -173,6 +173,10 @@ export default class Configuration {
             );
         }
 
+        if (config.serviceName === undefined || config.serviceName === null) {
```

Comment:
@tiffon is this the best way to check whether the serviceName was not set in node?

Reply:
* I think you want this at the very top of the function
 * can you not just check `if (!config.serviceName)`? An empty string as a service name is also unacceptable.